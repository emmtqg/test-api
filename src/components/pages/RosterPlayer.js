// @flow
import React from 'react';
import PropTypes from 'prop-types';
import TableRow from '@material-ui/core/TableRow';
import TableCell from '@material-ui/core/TableCell';
import Fab from '@material-ui/core/Fab';
import ClearIcon from '@material-ui/icons/Clear';
import { withStyles } from '@material-ui/core/styles';
import red from '@material-ui/core/colors/red';

import '../styles/SideBySideComponents.scss';

// Roster table row/icon styles
const styles = theme => ({
  tableCell: {
    fontSize: '1.6rem',
  },
  icon: {
    colorAction: red[500],
    margin: '0 auto',
  },
  iconHover: {
    color: red[500],
    margin: theme.spacing.unit * 2,
    '&:hover': {
      color: red[800],
    },
  },
  fab: {
    margin: theme.spacing.unit,
    backgroundColor: 'white',
    color: 'red',
  },
  extendedIcon: {
    marginRight: theme.spacing.unit,
  },
});

const RosterPlayer = ({ player, cbRemovePlayer, classes }) => {
  const {
    id,
    firstName,
    lastName,
    rating,
    handedness,
  } = player;

  const deleteClass = `${classes.icon} delete`;

  return (
    <TableRow key={id}>
      <TableCell align="left" className={classes.tableCell}>{firstName} {lastName}</TableCell>
      <TableCell align="right" className={classes.tableCell}>{rating}</TableCell>
      <TableCell align="right" className={classes.tableCell}>{handedness}</TableCell>
      <TableCell align="center" className={deleteClass}>
        <Fab
          aria-label="Delete"
          size="small"
          className={classes.fab}
          onClick={() => cbRemovePlayer(id)}
        >
          <ClearIcon />
        </Fab>
      </TableCell>
    </TableRow>
  );
};

RosterPlayer.defaultProps = {
  player: {},
};

RosterPlayer.propTypes = {
  player: PropTypes.shape({
    id: PropTypes.string.isRequired,
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    handedness: PropTypes.string.isRequired,
  }),
  // eslint-disable-next-line react/forbid-prop-types
  classes: PropTypes.object.isRequired,
  cbRemovePlayer: PropTypes.func.isRequired,
};

export default withStyles(styles)(RosterPlayer);
