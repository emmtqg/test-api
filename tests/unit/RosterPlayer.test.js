import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import RosterPlayer from '../../src/components/RosterPlayer';

let wrapper;

// auth mocks for Promise return of getPlayers API call from AuthHelpers.js
const player = {
  first_name: 'liam',
  last_name: 'neeson',
  rating: 9000,
  handedness: 'left',
  id: '5c9fdeee310d217ff3901e67',
};

// Mock functions for MUI classes and remove player dialog confirmation
const cbRemovePlayer = jest.fn(() => 'Generate confirm dialog');
const classes = {};
const props = { player, cbRemovePlayer, classes };

beforeAll(() => {
  wrapper = mount(<RosterPlayer {...props} />);
});

afterAll(() => {
  wrapper.unmount();
});

describe('The Roster Table Row (RosterPlayer) should', () => {
  it('render correctly (match snapshot)', () => {
    const component = renderer.create(<RosterPlayer {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('generate a table row element', () => {
    expect(wrapper.find('tr').length).toBe(1);
  });

  it('generate a table row for roster that includes player info and a delete option', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('generate a table cell with a delete icon and click callback', () => {
    const removeBtn = wrapper.find('button');
    expect(removeBtn.length).toBeGreaterThanOrEqual(1);
    expect(removeBtn.props('onClick')).toBeDefined();
    expect(typeof removeBtn.props('onClick').onClick === 'function').toBe(true);
  });
});
