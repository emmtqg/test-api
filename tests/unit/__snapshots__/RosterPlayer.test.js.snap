// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Roster Table Row (RosterPlayer) should generate a table row for roster that includes player info and a delete option 1`] = `"<tr class=\\"MuiTableRow-root-6\\"><td class=\\"MuiTableCell-root-11 MuiTableCell-alignLeft-19 RosterPlayer-tableCell-1\\">liam neeson</td><td class=\\"MuiTableCell-root-11 MuiTableCell-alignRight-21 RosterPlayer-tableCell-1\\">9000</td><td class=\\"MuiTableCell-root-11 MuiTableCell-alignRight-21 RosterPlayer-tableCell-1\\">left</td><td class=\\"MuiTableCell-root-11 MuiTableCell-alignCenter-20 RosterPlayer-icon-2 delete\\"><button class=\\"MuiButtonBase-root-33 MuiFab-root-23 MuiFab-sizeSmall-31 RosterPlayer-fab-4\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"Delete\\"><span class=\\"MuiFab-label-24\\"><svg class=\\"MuiSvgIcon-root-36\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root-45\\"></span></button></td></tr>"`;

exports[`The Roster Table Row (RosterPlayer) should render correctly (match snapshot) 1`] = `
<tr
  className="MuiTableRow-root-6"
>
  <td
    aria-sort={null}
    className="MuiTableCell-root-11 MuiTableCell-alignLeft-19 RosterPlayer-tableCell-1"
  >
    liam
     
    neeson
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root-11 MuiTableCell-alignRight-21 RosterPlayer-tableCell-1"
  >
    9000
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root-11 MuiTableCell-alignRight-21 RosterPlayer-tableCell-1"
  >
    left
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root-11 MuiTableCell-alignCenter-20 RosterPlayer-icon-2 delete"
  >
    <button
      aria-label="Delete"
      className="MuiButtonBase-root-33 MuiFab-root-23 MuiFab-sizeSmall-31 RosterPlayer-fab-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiFab-label-24"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-36"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-45"
      />
    </button>
  </td>
</tr>
`;
